[Doc] Streaming generator alpha doc (#39914)
ray-project/ray: doc/source/ray-core/tasks/generators.rst

.. _dynamic_generators:

.. warning::

>   ``num_returns="dynamic"`` :ref:`generator API <dynamic_generators>` is soft deprecated from Ray 2.8 due to its :ref:`limitation <dynamic-generators-limitation>` and hard deprecated from Ray 2.9.

Author: We have a task to deprecate this API in 2.8

>   ``num_returns="dynamic"`` :ref:`generator API <dynamic_generators>` is soft deprecated from Ray 2.8 due to its :ref:`limitation <dynamic-generators-limitation>` and hard deprecated from Ray 2.9.

Reviewer: ```suggestion
  ``num_returns="dynamic"`` :ref:`generator API <dynamic_generators>` is soft deprecated as of Ray 2.8 due to its :ref:`limitation <dynamic-generators-limitation>`. It is hard deprecated as of Ray 2.9.
```

>   Use a :ref:`streaming generator API<generators>` instead.

Reviewer: ```suggestion
  Use the :ref:`streaming generator API<generators>` instead.
```

> Dynamic Generators

Reviewer: ```suggestion
Dynamic generators
```

==================

> Python generators are functions that behave like an iterator, yielding one

Reviewer: ```suggestion
Python generators are functions that behave like iterators, yielding one
```
